azurelinuxagent/common/protocol/wire.py

```diff
@@ -961,7 +961,7 @@ def send_request_using_appropriate_channel(self, direct_func, host_func):
                              log_event=True)
                 raise
 
-        if not HostPluginProtocol.is_default_channel():
+        if ret and not HostPluginProtocol.is_default_channel():
```

Comment:
Main change of the PR. Previously, the host plugin channel would be set as default even if `ret` was None.